errors/middleware-upgrade-guide.md

```diff
@@ -0,0 +1,290 @@
+# Middleware Upgrade Guide
+
+As we work on improving Middleware for General Availability (GA), we've made some changes to the Middleware APIs (and how you define Middleware in your application) based on your feedback.
```

Comment:
I don't think the parentheses are necessary. Makes it a better easier to read without?
```suggestion
As we work on improving Middleware for General Availability (GA), we've made some changes to the Middleware APIs and to how you define Middleware in your application, based on your feedback.
```

Reply:
"to how you" feels strange - I'm open to dropping the parenthesis but might need a different revision.